services/s3/src/it/java/software/amazon/awssdk/services/s3/crossregion/S3CrossRegionIntegrationTestBase.java

```diff
@@ -0,0 +1,179 @@
+
+    protected static final Region CROSS_REGION = Region.of("eu-central-1");
+
+    private static final String KEY = "key";
+
+    protected CaptureInterceptor captureInterceptor;
+
+    @Test
+    void getApi_CrossRegionCall() {
+        s3.putObject(p -> p.bucket(bucketName()).checksumAlgorithm(ChecksumAlgorithm.CRC32).key(KEY), RequestBody.fromString(
+            "TEST_STRING"));
+        GetObjectRequest getObjectRequest =
+            GetObjectRequest.builder().bucket(bucketName()).checksumMode(ChecksumMode.ENABLED).key(KEY).build();
+        ResponseBytes<GetObjectResponse> response = getAPICall(getObjectRequest);
+        assertThat(regionsIntercepted(captureInterceptor.hosts())).isEqualTo(Arrays.asList(CROSS_REGION.id(),
+                                                                                           DEFAULT_REGION.id()));
+        assertThat(captureInterceptor.httpMethods()).isEqualTo(Arrays.asList(SdkHttpMethod.GET, SdkHttpMethod.GET));
+        assertThat(new String(response.asByteArray())).isEqualTo("TEST_STRING");
+        assertThat(captureInterceptor.getServiceCalls()).isEqualTo(2);
+    }
+
+    @Test
+    void putApi_CrossRegionCall() {
+        s3.putObject(p -> p.bucket(bucketName()).checksumAlgorithm(ChecksumAlgorithm.CRC32).key(KEY), RequestBody.fromString(
+            "TEST_STRING"));
+        PutObjectRequest putObjectRequest =
+            PutObjectRequest.builder().bucket(bucketName()).checksumAlgorithm(ChecksumAlgorithm.CRC32).key(KEY).build();
+        PutObjectResponse response = putAPICall(putObjectRequest, "TEST_STRING");
+        assertThat(regionsIntercepted(captureInterceptor.hosts())).isEqualTo(Arrays.asList(CROSS_REGION.id(),
+                                                                                           DEFAULT_REGION.id()));
+        assertThat(captureInterceptor.httpMethods()).isEqualTo(Arrays.asList(SdkHttpMethod.PUT, SdkHttpMethod.PUT));
+        assertThat(response.checksumCRC32()).isEqualTo("S9ke8w==");
+        assertThat(captureInterceptor.getServiceCalls()).isEqualTo(2);
+    }
+
+    @Test
+    void deleteApi_CrossRegionCall() {
+        s3.putObject(p -> p.bucket(bucketName()).checksumAlgorithm(ChecksumAlgorithm.CRC32).key(KEY), RequestBody.fromString(
+            "TEST_STRING"));
+        DeleteObjectRequest deleteObjectRequest = DeleteObjectRequest.builder().bucket(bucketName()).key(KEY).build();
+        DeleteObjectResponse response = deleteObjectAPICall(deleteObjectRequest);
+        assertThat(regionsIntercepted(captureInterceptor.hosts())).isEqualTo(Arrays.asList(CROSS_REGION.id(),
+                                                                                           DEFAULT_REGION.id()));
+        assertThat(captureInterceptor.httpMethods()).isEqualTo(Arrays.asList(SdkHttpMethod.DELETE, SdkHttpMethod.DELETE));
+        assertThat(response).isNotNull();
+        assertThat(captureInterceptor.getServiceCalls()).isEqualTo(2);
```

Comment:
This seems like it would be flaky if there are ever any retries. Is there some way to rewrite the checks so that it's more resilient to retries?

Reply:
Good catch.
Removed the interceptors
Testing using interceptors is covered in Junits.
Now I just make sure the responses from Cross region calls are as expected